### ISSUE TITLE
Update Tensorflow Stub DType

### ISSUE THREAD

**Issue by j3soon** [tensorflow/tensorboard#6899]
## Environment information (required)

Not related.

## Issue description

The tensorflow stub [`tensorboard/compat/tensorflow_stub/dtypes.py`](https://github.com/tensorflow/tensorboard/blob/master/tensorboard/compat/tensorflow_stub/dtypes.py) does not include certain types in the latest tensorflow [`tensorflow/python/framework/dtypes.py`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/framework/dtypes.py). Such as:

- `types_pb2.DT_FLOAT8_E5M2_REF`
- `types_pb2.DT_FLOAT8_E4M3FN_REF`
- `types_pb2.DT_INT4_REF`
- `types_pb2.DT_UINT4_REF`

These types exist in [`tensorboard/compat/proto/types.proto`](https://github.com/tensorflow/tensorboard/blob/ce8920935c5b14d1558947f78b5e45b22aed3896/tensorboard/compat/proto/types.proto#L79-L86) and in [`tensorflow/python/framework/dtypes.py`](https://github.com/tensorflow/tensorflow/blob/4a4da9ee4117e19727ef3ddd0e288114e85ae5de/tensorflow/python/framework/dtypes.py#L506-L509).

I can open a PR to update the `dtypes.py` file according to the latest tensorflow code. However, I think it will require adding `ml_dtypes` as an extra dependency, and I'm not sure if this is the preferred approach.

**Comment by arcra** [comment]
Hmmm... I don't know if we want to include this dependency at this point. Honestly, I'm not well aware at this time of how this impacts users.

Those types seem to be [noted as "experimental"](https://github.com/tensorflow/tensorflow/blob/f8bffb7433b1083aeda64095eb9e38e6a2fc0a55/tensorflow/python/framework/dtypes.py#L456) in some way.

Have you experienced any issues with the tool because of this? Or was this mostly just an FYI?

**Comment by j3soon** [comment]
Hi @arcra, I haven't encountered any issues with tensorboard, but I noticed that the stubs have significantly diverged from the main TensorFlow repository. Do you think we should update them now, or would it be better to resolve this when something breaks in the future?

**Comment by arcra** [comment]
Yes, honestly, we don't have bandwidth at the moment, and it's unclear if users are using these, and to what extent other parts of the code need to be updated. It looks like our `tensor_util.py` is the piece of code that would care about this, but we'd need to spend some time understanding if we want to and how to add support for these types there as well.

So I'd punt it until someone asks for support for this, but we can keep this open for now. Thanks for letting us know.
